src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -126,6 +127,14 @@ public GitHub github() {
     return github;
   }
 
+  /**
+   * Set the interface to the GitHub API.
+   */
+  public void github(GitHub github) {
```

Comment:
Could you please rename it to `set()`? I think that the `github` field needs also be synchronized.

Reply:
I wanted to name it `setGithub` but after I have seen the `github()` method without `get...` I thought without `set` is preferred 🙂
Regarding synchronized for the github field I guess I do not understand it.
Would it make a real difference if the get/set for `github` have a synchronized block?

Reply:
> I wanted to name it `setGithub` but after I have seen the `github()` method without `get...` I thought without `set` is preferred 

I meant `public void set(GitHub github) {`.

> Regarding synchronized for the github field I guess I do not understand it.
> Would it make a real difference if the get/set for `github` have a synchronized block?

Since the `github` field is not final anymore, multiple threads can read and write to the `github` field. Therefore, all access to this field needs to be synchronized.

Reply:
Still not sure if this would make a bit difference.
But I will update it as proposed and for a detailed explanation we use another channel 🙂

Reply:
Most likely you won't see any difference. If `github` is not synchronized, and you use multiple instances of `GitHub`, then you can't tell which one is used for sure once you update the field. I would like to make the class completely thread-safe.

Reply:
Still not sure about this. 
But I trust you as you have more knowledge about concurrency than I 🙂